lib/src/api.dart

```diff
@@ -37,19 +37,16 @@ abstract class Serializer {
   static Object fromJson(String json, [Type type]) => _fromJson(json, type);
 
   /// Convert a Map<String, dynamic>
-  static Object fromMap(Map map, [Type type]) => _fromMap(map, type);
+  static Object fromMap(Map map, [Type type, List<Type> mapOf]) => _fromMap(map, type, mapOf);
```

Comment:
Ok so maybe you can check if the first is a String, add it if not or throw an error